src/importlinter/domain/imports.py

```diff
@@ -60,29 +63,29 @@ def is_package(self) -> bool:
         raise NotImplementedError
 
 
-class DirectImport(ValueObject):
+class ImportExpression(ValueObject):
     """
-    An import between one module and another.
+    A python import expression.
```

Comment:
That's not quite right. I would define this as something like:

```
An expression describing an import or set of imports.

Sets of imports are notated using * wildcards. These wildcards
can stand in for a module name or part of a name, but they do
not extend to subpackages. 

For example, "mypackage.*" refers to every child subpackage
of mypackage. It does not, however, include more distant
descendants such as mypackage.foo.bar. 
```

Note I've also tried to define more precisely what the wildcard denotes. Open to other options but whatever we choose should be documented here.

Reply:
The wildcard in fact acts on the string level meaning that `x.*.bar` matches `x.foo.bar` but also `x.foo.dummybar`

Reply:
So, as written above, let me rewrite this. However, it make sense that `module.*` matches `module.foo` but also `module.foo.bar`.

Reply:
It feels counterintuitive to me. I can imagine that users are more likely to want to have more control, but open to persuasion. In your particular use case, would it make a difference either way?

> Else we need to make a compare per module level making this very complex

I'm not sure it would be _that_ complex. [find_children](https://grimp.readthedocs.io/en/stable/usage.html#ImportGraph.find_children) could be used to drill down and compare each module name. Happy to help if you get stuck.

Reply:
Done it with regex. fnmatch does by the way nothing else